test/index.test.ts

```diff
@@ -3256,25 +3256,31 @@ describe('Unit | Utility | changeset', () => {
     });
   });
 
-  it('#restore restores a snapshot of the changeset with nested values', () => {
-    let user = { name: 'Adam', address: { country: 'United States' } };
+  it('#restore restores a snapshot of the changeset when nested value is object', () => {
```

Comment:
@sapryniukt could you please keep the existing test case as is and only add a new test?

cc @snewcomer this test mimics the scenario that fails as today `restore` would not honor an instance of the `Country` class and would only restore the object structure

Reply:
Sorry for my lack of familiarity but I’m guessing we can’t fix the bug by keeping the existing data shape?